src/chatClient.cpp

```diff
@@ -2437,7 +2441,18 @@ void ContactList::syncWithApi(mega::MegaUserList& users)
 
 void ContactList::onUserAddRemove(mega::MegaUser& user)
 {
-    addUserFromApi(user);
+    if (user.getVisibility() == ::mega::MegaUser::VISIBILITY_INACTIVE)
```

Comment:
Inactive users should be effectively removed. We want to keep `HIDDEN` users, which mean ex-contact but still users of MEGA. An `INACTIVE` user is the one that closed his account, and therefore it's deleted from every record in the SDK (and it should do the same in karere). The `userid` becomes invalid too.